### ISSUE TITLE
5.0.0 execute in parallel broken

### ISSUE THREAD

**Issue by skywalkerdude** [Quick/Quick#1139]
- [ x] I have read [CONTRIBUTING](https://github.com/Quick/Quick/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Related to https://github.com/Quick/Quick/issues/1123, but I updated my library to version 5.0.0 and tried running my Quick tests with "execute in parallel" on.

### What did you expect to happen?

Tests would run

### What actually happened instead?

Tests failed with error
```
 Assertions: System: Unexpected TestCaseDidStart. If you believe this error represents a bug, please attach the result bundle at ...DerivedData/Sample-acinzmmaeypryxgcsczdnkgqvssb/Logs/Test/Test-Sample-2022.04.16_00-09-09--0700.xcresult
```
<img width="1512" alt="Screen Shot 2022-04-16 at 12 14 01 AM" src="https://user-images.githubusercontent.com/1427524/163665679-4db510f0-07c4-47de-9ed5-916124f97427.png">

### Environment

List the software versions you're using:

 - Quick: *5.0.0*
 - Nimble: *N/A*
 - Xcode Version: *13.3*
 - Swift Version: `Xcode Default`

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - CocoaPods: *1.11.3*
 - Carthage: *N/A*
 - Swift Package Manager: *N/A*

### Project that demonstrates the issue
[Sample.zip](https://github.com/Quick/Quick/files/8499458/Sample.zip)

**Comment by NachoSoto** [comment]
Same here:
<img width="475" alt="Screen Shot 2022-04-16 at 15 04 25" src="https://user-images.githubusercontent.com/685609/163676059-dfe16015-28a2-421d-bee0-06bd142518bd.png">

**Comment by younata** [comment]
This is also an issue in Xcode 13.2 at least.

This is likely the result of #1129 solving the issue in an incomplete way. I'll investigate it some more this evening.

**Comment by jessesquires** [comment]
Fixed in #1143
